src/backend/access/appendonly/appendonly_visimap_store.c

```diff
@@ -71,7 +70,7 @@ AppendOnlyVisimapStore_Init(AppendOnlyVisimapStore *visiMapStore,
 	Assert(OidIsValid(visimapRelid));
 	Assert(OidIsValid(visimapIdxid));
 
-	visiMapStore->snapshot = RegisterSnapshot(snapshot);
```

Comment:
If we don't register snapshot, when and where the snapshot is freed?

Reply:
I'm sorry to answer your question for so long. For unique check, the snapshot is allocate and destory in the stack in the fuction _bt_check_unique, see nbtinsert.c:431(InitDirtySnapshot(SnapshotDirty);) for detail